src/storage-backend.js

```diff
@@ -0,0 +1,655 @@
+let assert = require('assert');
+let crypto = require('crypto');
+let url = require('url');
+let debug = require('debug')(require('path').relative(process.cwd(), __filename));
+let http = require('http');
+
+// We don't follow redirects inside the request library because it does not
+// have a guaruntee that all redirects are secure.  This is a critical part of
+// our system, so we've implemented our own secure redirector
+let request = require('request').defaults({
+  followRedirect: false, 
+});
+let fs = require('fs');
+let stream = require('stream');
+let meter = require('stream-meter');
+let SQSConsumer = require('sqs-consumer');
+
+
+
+/**
+ * Obtain a header from a URL.  Do not follow redirects.  Returns the headers,
+ * a caseless to look up header names without worrying about case, the HTTP
+ * status code and the HTTP status message
+ */
+let requestHead = async (u) => {
+  return new Promise((resolve, reject) => {
+    request.head(u, {followRedirect: false}, (err, response) => {
+      if (err) {
+        debug(err.stack || err);
+        return reject(err);
+      }
+      debug('got headers');
+      resolve({
+        headers: response.headers,
+        caseless: response.caseless,
```

Comment:
Pretty sure `res.headers` are supposed to be lowercase... Maybe requests goes out of it's way to do the wrong thing...
But `http/https` always does this right. Ie. gives you lowercase headers.